### DISCUSSION TITLE
grains segmentation

### DISCUSSION THREAD

**Discussion by makeabhishek** [mtex-toolbox/mtex#1462] [Ask Anything]
Dear All,
First of all thanks for providing the mtex repository. It has made many tasks easy.
I want to generate a synthetic microstructure similar to a given microstructure, maintaining the grain size distribution and texture of the material.
(1) To achieve that I have to segment grains into large and small features from an EBSD file (.ang, or .csv file), attached is the EBSD image and obtain the statistics. But prior to that, I also need to smooth the regions (example: as marked in the attached image by the black boxes) and fill the unindex points. How can I achieve this? I tried calculating grains based on the threshold 0f 15 to 30 degree but it doesn't seem to work well.

Furthermore,  I also want to keep the texture components the similar to original EBSD. I tried `ori = discreteSample(odf,npoints)` , which can generate random points. But how do keep the grain ID's in such case to allocate texture values or Euler angles to the same grains which were segmented in step (1).

![ebsd](https://user-images.githubusercontent.com/47937684/174766760-ac437b30-f96c-41bd-a3dd-4455758afa41.png)

**Comment by lfgmorales** [comment]
Hi

I am not entirely sure I understood the question, but are the red-yellow domains elongated grains, and you want to have a synthetic microstructure with those?

Maybe you can start by cleaning up your data (the map you provided is pretty noisy), then demonising it (https://mtex-toolbox.github.io/EBSDDenoising.html <https://mtex-toolbox.github.io/EBSDDenoising.html>), then calculate grains. If the red bands are single grains, 10° should do it I would say. But if the noisy data, playing with the disorientation degree won’t change much

cheers


Luiz
